### PR TITLE
feat: Add google sheets

### PR DESCRIPTION
**Pull request by karthik-sivadas**
## Description
feat: Add google sheets

## Related Issue
https://github.com/Multiwoven/multiwoven-integrations/issues/86

## Type of Change
- [x] New Connector (Destination or Source Connector)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would impact existing functionality)
- [ ] Documentation update
- [ ] Chore
  
## How Has This Been Tested?
Wrote unit test and tested locally by sending data to google sheets

## Checklist:
- [x] Ensure a branch name is prefixed with `feature`, `bugfix`, `hotfix`, `release` or `chore` followed by `/` and branch name e.g `feature/add-salesforce-connector`
- [x] Added the new connector in rollout.rb
- [x] Have you updated the version number of the gem?
- [x] Have you ensured that your changes for new connector are documented in the [docs repo](https://github.com/Multiwoven/docs) or relevant documentation files?
- [x] Have you updated the run time dependency in multiwoven-integrations.gemspec if new gems are added

### CODE REVIEW

File: lib/multiwoven/integrations/destination/google_sheets/client.rb [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+
+          def generate_header_range(sheet_name, last_column_index)
+            "#{sheet_name}!A1:#{column_index_to_letter(last_column_index)}1"
+          end
+
+          def column_index_to_letter(index)
+            ("A".."ZZZ").to_a[index - 1]
+          end
+
+          def generate_json_schema(column_names, sheet_name)
```
Comment (afthabvp) [inline lib/multiwoven/integrations/destination/google_sheets/client.rb:94]: The MAX_CHUNK_SIZE variable is set to 10,000, indicating the maximum number of records that can be processed in a single batch.
 If Google Sheets supports batch operations add these fields in your stream
 
 `   "batch_support": true,
      "batch_size": 10000,`

File: lib/multiwoven/integrations/destination/google_sheets/client.rb [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+            {
+              name: sheet_name,
+              action: "create",
+              json_schema: generate_properties_schema(column_names),
+              supported_sync_modes: %w[full_refresh incremental]
```
Comment (afthabvp) [inline lib/multiwoven/integrations/destination/google_sheets/client.rb:99]: full_refresh not yet implemented
please remove it

File: lib/multiwoven/integrations/destination/google_sheets/client.rb [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+module Multiwoven
+  module Integrations
+    module Destination
+      module GoogleSheets
+        include Multiwoven::Integrations::Core
+
+        class Client < DestinationConnector # rubocop:disable Metrics/ClassLength
+          MAX_CHUNK_SIZE = 10_000
+          GOOGLE_SHEETS_SCOPE = "https://www.googleapis.com/auth/drive"
+          GOOGLE_SHEETS_SCHEMA_URL = "http://json-schema.org/draft-07/schema#"
```
Comment (afthabvp) [inline lib/multiwoven/integrations/destination/google_sheets/client.rb:12]: It's a JSON schema URL. Move it to the constants so we can use it.

**Review by subintp — CHANGES_REQUESTED** [review]

File: lib/multiwoven/integrations/destination/google_sheets/config/meta.json [hunk 4]
```diff
@@ -0,0 +1,15 @@
+{
+  "data": {
+    "name": "GoogleSheets",
+    "title": "Google Sheets",
+    "connector_type": "destination",
+    "category": "Productivity Tools",
+    "documentation_url": "https://docs.multiwoven.com/destinations/productivity-tools/google-sheets-service-account",
```
[the remark this answers is not in the document]
Comment (karthik-sivadas) [reply lib/multiwoven/integrations/destination/google_sheets/config/meta.json:7]: Have created a draft PR need minor changes. Will update it. The link will be same.

File: lib/multiwoven/integrations/destination/google_sheets/client.rb [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+
+module Multiwoven
+  module Integrations
+    module Destination
+      module GoogleSheets
+        include Multiwoven::Integrations::Core
+
+        class Client < DestinationConnector # rubocop:disable Metrics/ClassLength
+          MAX_CHUNK_SIZE = 10_000
+          GOOGLE_SHEETS_SCOPE = "https://www.googleapis.com/auth/drive"
```
[the remark this answers is not in the document]
Comment (karthik-sivadas) [reply lib/multiwoven/integrations/destination/google_sheets/client.rb:11]: We need to pass the scope for google API. I will add a comment with link

**Review by subintp — APPROVED** [review]

**Pull request merged**